rq/job.py

```diff
@@ -733,4 +732,43 @@ def register_dependency(self, pipeline=None):
             connection.sadd(dependents_key, self.id)
             connection.sadd(self.dependencies_key, dependency_id)
 
+    def get_dependencies_statuses(
+            self,
+            watch=False,
+            pipeline=None
+    ):
+        """Returns a list of tuples containing the job ids and status of all
+        dependencies; e.g:
+
+           [('14462606-09c4-41c2-8bf1-fbd109092318', 'started'),
+            ('e207328f-d5bc-4ea9-8d61-b449891e3230', 'finished'),  ...]
+
+        As a minor optimization allowing callers to more quickly tell if all
+        dependencies are _FINISHED_, the returned list is sorted by the
+        `ended_at` timestamp, so those jobs which are not yet finished are at
+        the start of the list.
+        """
+
+        pipe = pipeline if pipeline is not None else self.connection
+
+        if watch:
+            pipe.watch(self.dependencies_key)
```

Comment:
Do we need to watch `dependencies_key`? If I'm not mistaken, this key contains a list of dependencies, which should not change after the job is first enqueued.

Reply:
ah yes - perhaps a bit too defensive here. Probably better to not make the unnecessary call.

Reply:
@selwin in f624ab78432299c8e021604b40636675df2da341